cpp/cgi_progs/zotero_cgi.cc

```diff
@@ -436,16 +440,21 @@ bool ProcessShowQASubActionDelete(const std::multimap<std::string, std::string>
 {
     const std::string delete_type(GetCGIParameterOrDefault(cgi_args, "delete_type"));
     const std::string delete_tag(GetCGIParameterOrDefault(cgi_args, "delete_tag"));
-    if (delete_type.empty() or delete_tag.empty())
+    const std::string delete_subfield_code(GetCGIParameterOrDefault(cgi_args, "delete_subfield_code", ""));
+    const std::string delete_record_type(GetCGIParameterOrDefault(cgi_args, "delete_record_type"));
+    if (delete_type.empty() or delete_tag.empty() or delete_subfield_code.empty() or delete_record_type.empty())
         return false;
 
     std::string journal_id_to_delete(" = " + db_connection->escapeAndQuoteString(zeder_journal_id));
     if (delete_type == "global")
         journal_id_to_delete = "IS NULL";
 
     db_connection->queryOrDie("DELETE FROM metadata_presence_tracer "
-                             "WHERE zeder_journal_id " + journal_id_to_delete + " "
-                             "AND metadata_field_name = " + db_connection->escapeAndQuoteString(delete_tag));
+                             "WHERE zeder_journal_id = " + journal_id_to_delete +
+                             " AND metadata_field_name = " + db_connection->escapeAndQuoteString(delete_tag) +
+                              " AND subfield_code = " +
```

Comment:
- indentation
- `= NULL` is valid SQL syntax but has a different semantic meaning and does not get you the correct values, you need to use `IS NULL`